ambari-server/src/main/java/org/apache/ambari/server/controller/internal/BlueprintResourceProvider.java

```diff
@@ -175,13 +184,14 @@
    * Static initialization.
    *
    * @param factory   blueprint factory
-   * @param dao       blueprint data access object
+   * @param bpDao       blueprint data access object
    * @param gson      json serializer
    */
-  public static void init(BlueprintFactory factory, BlueprintDAO dao, SecurityConfigurationFactory
-    securityFactory, Gson gson, AmbariMetaInfo metaInfo) {
+  public static void init(BlueprintFactory factory, BlueprintDAO bpDao, TopologyRequestDAO trDao,
```

Comment:
I know it was already like this, but how difficult would it be to get rid of init() and use StaticallyInject for this?

Reply:
I think it is as this class is initialized in ambari srver after JPA service startup while static injections happen in the constructor of the controller module

Reply:
`init()` has been replaced with regular injection on `branch-feature-AMBARI-14174`:

https://github.com/apache/ambari/blob/01c9e4bd5d2e15fd842bb4ad88a2e2c52af55def/ambari-server/src/main/java/org/apache/ambari/server/controller/internal/BlueprintResourceProvider.java#L186-L194

I think we should keep it on trunk to minimize merge conflict.